src/modules/Channel/components/SuggestedMentionList/index.tsx

```diff
@@ -93,43 +110,72 @@ function SuggestedMentionList(props: SuggestedMentionListProps): JSX.Element {
+    if (lastSearchString && searchString.indexOf(lastSearchString) === 0 && currentMemberList.length === 0) {
+      // Don't need to request query again
       return;
     }
+    // Add member list query for customization
+    if (!channelInstance.isSuper) {
+      const suggestingMembers = getCurrentSearchedMemberList();
+      if (suggestingMembers.length < 1) {
+        logger.info('SuggestedMentionList: Channel\'s member list is empty');
+      } else {
+        logger.info('SuggestedMentionList: Channel\'s member list has been updated.', { memberList: suggestingMembers });
+        setCurrentUser(suggestingMembers[0]);
+      }
+      setLastSearchString(searchString);
+      onFetchUsers(suggestingMembers);
+      setCurrentMemberList(suggestingMembers);
+    }
+  }, [
+    channelInstance?.url,
+    searchString,
+    ...currentSearchedMemberList,
+  ]);
+
+  /* Fetch member list for super channel */
+  useEffect(() => {
     if (lastSearchString && searchString.indexOf(lastSearchString) === 0 && currentMemberList.length === 0) {
       // Don't need to request query again
       return;
     }
-
-    const query = channelInstance?.createMemberListQuery({
-      limit: maxSuggestionCount + 1, // because current user could be included
-      nicknameStartsWithFilter: searchString.slice(USER_MENTION_TEMP_CHAR.length),
-    });
     // Add member list query for customization
-    query.next()
-      .then((memberList) => {
-        const suggestingMembers = memberList
-          .filter((member) => currentUserId !== member?.userId)
-          .slice(0, maxSuggestionCount);
-        if (suggestingMembers.length < 1) {
-          logger.info('SuggestedMentionList: Fetched member list is empty');
-        } else {
-          logger.info('SuggestedMentionList: Fetching member list succeeded', { memberListQuery: query, memberList: suggestingMembers });
-          setCurrentUser(suggestingMembers[0]);
-        }
-        setLastSearchString(searchString);
-        onFetchUsers(suggestingMembers);
-        setCurrentMemberList(suggestingMembers);
-      })
-      .catch((error) => {
-        if (error) {
-          logger.error('SuggestedMentionList: Fetching member list failed', error);
-        }
+    if (channelInstance.isSuper) {
+      if (!channelInstance?.createMemberListQuery) {
+        logger.warning('SuggestedMentionList: Creating member list query failed');
+        return;
+      }
+      const query = channelInstance?.createMemberListQuery({
+        limit: maxSuggestionCount + 1, // because current user could be included
+        nicknameStartsWithFilter: searchString.slice(USER_MENTION_TEMP_CHAR.length),
       });
-  }, [channelInstance?.url, searchString]);
+      query.next()
+        .then((memberList) => {
+          const suggestingMembers = memberList
+            .filter((member) => currentUserId !== member?.userId)
+            .slice(0, maxSuggestionCount);
```

Comment:
How about creating a function to handle and manage the part that retrieves users?

```ts
async function fetchMembersFromChannel(channel, maxSuggestionCount, searchString): Promise<Member[]> {
    return channel.members
      .sort((a, b) => a.nickname?.localeCompare(b.nickname))
      .filter(
        (member) => member.nickname?.toLowerCase().startsWith(searchString.toLowerCase())
        && member.userId !== currentUser?.userId
        && member.isActive,
      ).slice(0, maxSuggestionCount);
}

async function fetchMembersFromQuery(channel, maxSuggestionCount, searchString): Promise<Member[]> {
    const query = channel.createMemberListQuery({
      limit: maxSuggestionCount + 1, // because current user could be included
      nicknameStartsWithFilter: searchString.slice(USER_MENTION_TEMP_CHAR.length),
    });

    return query.next()
      .then((memberList) => {
        return memberList
          .filter((member) => currentUserId !== member?.userId)
          .slice(0, maxSuggestionCount);
      }
}
```

Reply:
Thanks for the great suggestion. I have applied this change but the dependency I used is a bit different.
```[
    channelInstance?.url,
    // We have to be specific like this or React would not recognize the changes in instances.
    channelInstance.members.map((member: Member) => member.userId).join(),
    channelInstance.members.map((member: Member) => member.nickname).join(),
    channelInstance.members.map((member: Member) => member.isActive).join(),
    searchString,
    maxSuggestionCount,
    currentUserId,
    currentMemberList,
    lastSearchString
  ]
```
let me know if this looks ok for you.